src/components/HomeSlider.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
+
+const HomeSlider = ({ banners }) => {
+  const [current, setCurrent] = useState(0);
+  const length = banners.length;
+
+  const nextSlide = () => {
+    setCurrent(current === length - 1 ? 0 : current + 1);
+  };
+
+  if (!Array.isArray(banners) || banners.length <= 0) {
+    return null;
+  }
```

Comment:
We can avoid this check by using [prop types](https://reactjs.org/docs/typechecking-with-proptypes.html), btw map over an empty array does not throw an error or something, it just does not render anything.